### ISSUE TITLE
Support for direct-to-browser ESM

### ISSUE THREAD

**Issue by MagnusOxlund** [EvanHahn/HumanizeDuration.js#225]
From the docs, it looks like the project installs in the browser's global namespace.

Are there any plans to distribute an ES module directly to browsers (without Node.js) that can be imported with an ES6 `import` declaration?

For instance:
```html
<!DOCTYPE html>
<html lang="en">
<head>
  <link rel="modulepreload" href="https://path/to/HumanizeDuration.js">
  <script type="module">
    import humanizeDuration from 'https://path/to/HumanizeDuration.js';
    console.log(humanizeDuration(97320000));
  </script>
</head>
</html>
```

**Comment by MagnusOxlund** [comment]
For reference, [Luxon](https://moment.github.io/luxon/#/install?id=es6) is distributed in that form, with both a full and a minified version available.

**Comment by EvanHahn** [comment]
I'd like to add ES module support to this package, but I'm not sure how I want to do that yet.

I've added it to my to-do list. I will probably not get to this soon, apologies.

**Comment by The-LukeZ** [comment]
> but I'm not sure how I want to do that yet.

Could I make PR if?

**Comment by EvanHahn** [comment]
Yes, but please let me know how you plan to do this beforehand. There are a lot of different ways to accomplish this and I don't want you to have to do a bunch of unnecessary work.

**Comment by The-LukeZ** [comment]
> Yes, but please let me know how you plan to do this beforehand. There are a lot of different ways to accomplish this and I don't want you to have to do a bunch of unnecessary work.

Okay, sorry for the late response, I had to finish a personal project 😅

__I would fork this repo and do the following things:__
- make **`d.ts` file** for all type declarations (all `@typedef` -> `interface` and `type`)
- use the common **ES2016** or **ES2020** syntax (import instead of require) - tell what I should use: **ES2016** or **ES2020**
- create a default export for the default function - which I also want to deconstruct a bit, if possible

**Comment by EvanHahn** [comment]
Thanks for writing this out, because it helps me figure out what's needed.

I think my requirements are:

- Add ES Module compatibility.
- Maintain ECMAScript 3 compatibility. This module is still compatible with very old browsers, and I'd like to keep it that way.
- Maintain CommonJS support.
- Maintain AMD support (see [the code](https://github.com/EvanHahn/HumanizeDuration.js/blob/2082b0b70a596b6c24583a9545202825fcf83fc2/humanize-duration.js#L1950-L1954)).
- Maintain `window` support. It should still be possible to use this module with CommonJS, and it should still be possible to use `<script src="humanize-duration.js"></script>`. Importantly, it should still be possible to copy `humanize-duration.js` and use it in a `<script>` tag with no changes.
- Do not change TypeScript support as part of this work. It's *possible* we want to add that, but that's unrelated to ESM.

I think that means shipping two files in the npm package: `humanize-duration.cjs` and `humanize-duration.mjs` (or something like that). The latter would be auto-generated and put in the `.gitignore` file, and only included at build time.

Is that something you could help with?

**Comment by sbaechler** [comment]
No worries, I am not using the library either 😂. The calculation is using constant values and does not account for different lengths in months or daylight saving time.

Date-fns has a function that converts Durations in human readable form given a start date. I think the new [Javascript Temporal](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Temporal/Duration) object also supports this natively.

**Comment by The-LukeZ** [comment]
@sbaechler oh, interesting 👀

**Comment by EvanHahn** [comment]
Because this library is in maintenance mode, I'm not going to add this feature. If someone would like to fork it and add these features, I'm happy to add a link in the README.

**Comment by sbaechler** [comment]
Temporal is now Stage 4, so I guess we can use a browser native API for this going forward.

[Temporal.Duration.toLocaleString()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Temporal/Duration/toLocaleString)

```
const duration = Temporal.Duration.from({ hours: 1, minutes: 30, seconds: 15 });

console.log(duration.toLocaleString()); // 1 hr, 30 min, 15 sec
```
